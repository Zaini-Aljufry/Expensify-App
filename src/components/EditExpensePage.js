import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { setEditExpense, setRemoveExpense } from '../actions/expenses';

export class EditExpensePage extends React.Component{
  onSubmit =(expense)  => {
    this.props.setEditExpense(this.props.expense.id, expense);
    this.props.history.push('/dashboard');
  }
  onRemove = () => {
    this.props.setRemoveExpense({ id: this.props.expense.id });
    this.props.history.push('/dashboard');
  }
  render() {
    return(
      <div>
      <div className="page-header">
      <div className="content-container">
        <h1 className="page-header__title">Edit Expense</h1>
      </div>
      </div>
      <div className="content-container">
          <ExpenseForm
          expense={this.props.expense}
          onSubmit={this.onSubmit}
          />
          <button className="button button--sec" onClick={this.onRemove}>Remove Expense</button>
      </div>
    
    </div>
    )
  }
}


const mapDispatchToProps = (dispatch, props) =>({
  setEditExpense: (id,expense) => dispatch(setEditExpense(id,expense)),
  setRemoveExpense: (data) => dispatch(setRemoveExpense(data))
})


const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
  };
};

export default connect(mapStateToProps,mapDispatchToProps)(EditExpensePage);



// const EditExpensePage = (props) => {
//   return (
//     <div>
//       <ExpenseForm
//         expense={props.expense}
//         onSubmit={(expense) => {
//           props.dispatch(editExpense(props.expense.id, expense));
//           props.history.push('/');
//         }}
//       />
//       <button onClick={() => {
//         props.dispatch(removeExpense({ id: props.expense.id }));
//         props.history.push('/');
//       }}>Remove</button>
//     </div>
//   );
// };
