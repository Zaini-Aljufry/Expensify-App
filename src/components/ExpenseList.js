import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
  <div>
    {
      props.expenses.length === 0 ? (
        <h1>No Expenses</h1>
      ): (
        <div>
           <h1>Expense List</h1>
           {props.expenses.map((expense) => {
              return <ExpenseListItem key={expense.id} {...expense} />;
            })}
        </div>
      )
    }
    
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpenseList);
