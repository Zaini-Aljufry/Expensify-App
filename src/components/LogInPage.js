import React from 'react';
import { connect } from 'react-redux';
import {startLogin} from '../actions/auth'

export const LogInPage = ({startLogin}) => {
        return(
            <div>
                <h1>Log In Page</h1>
                <button onClick={startLogin}>Log In</button>
            </div>
        )
}

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
})


export default connect(undefined,mapDispatchToProps)(LogInPage);