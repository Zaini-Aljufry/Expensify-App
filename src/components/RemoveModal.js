import React from 'react'
import Modal from 'react-modal'

const RemoveModal = (props) => (
    <Modal
        isOpen={!!props.showModal}
        contentLabel="Remove modal"
        onRequestClose={props.handleCloseModal}
        closeTimeoutMS={200}
        className="modal"
    >
        <h3 className="modal__title">Remove Expense?</h3>
        <p>{props.body}</p>
        <button className="button button--modal" onClick={props.handleConfirm} >Okay</button>
        <button className="button button--modal" onClick={ props.handleCloseModal }>{ props.cancelText }</button>
    </Modal>
)

module.exports = {
    RemoveModal
}