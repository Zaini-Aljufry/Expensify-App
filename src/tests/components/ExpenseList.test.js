import {shallow} from 'enzyme'
import React from 'react';
import {ExpenseList} from '../../components/ExpenseList'
import expenses from '../fixtures/expenses'
import toJSON from 'enzyme-to-json'//serializer


test('should render ExpenseList with expenses correctly', () => {
    const wrapper = shallow(<ExpenseList expenses={expenses}/>)
    expect(toJSON(wrapper)).toMatchSnapshot();
})

test('should render ExpenseList without expenses correctly', () => {
    const empty = []
    const wrapper = shallow(<ExpenseList expenses={empty}/>)
    expect(toJSON(wrapper)).toMatchSnapshot();
})
