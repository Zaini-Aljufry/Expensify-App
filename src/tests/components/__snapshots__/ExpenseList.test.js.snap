// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList with expenses correctly 1`] = `
<div>
  <div>
    <h1>
      Expense List
    </h1>
    <ExpenseListItem
      amount="5000"
      createdAt={0}
      description="Nasi Goreng"
      id="1"
      key="1"
      note="My lunch"
    />
    <ExpenseListItem
      amount="5500"
      createdAt={-345600000}
      description="Beef Fried Rice"
      id="2"
      key="2"
      note="My lunch"
    />
    <ExpenseListItem
      amount="1600"
      createdAt={345600000}
      description="Kopi Peng"
      id="3"
      key="3"
      note="Drinks"
    />
  </div>
</div>
`;

exports[`should render ExpenseList without expenses correctly 1`] = `
<div>
  <h1>
    No Expenses
  </h1>
</div>
`;
